### PR TITLE
Fix assorted issues

### PR DESCRIPTION
**Pull request by geom3trik**
- Expose `Key`
- Add `set_hover` method to context
- Add bool to textbox `on_submit` to distinguish between enter and defocus
- Scale border by scale factor for layout

### CODE REVIEW

**Comment by robbert-vdh** [comment]
Forgot to post here, but this together with #160 fixes all of the issues I had with with the vizia GUIs in NIH-plug.

File: core/src/views/scrollview.rs [hunk 1]
```diff
@@ -107,10 +107,15 @@ impl<L: Lens<Target = ScrollData>> ScrollView<L> {
...
-                handle.left(Units::Pixels(-left)).top(Units::Pixels(-top));
+                let dpi_factor = handle.cx.style().dpi_factor;
+                if dpi_factor > 0.0 {
+                    let data = data.get(handle.cx);
+                    let left = ((data.child_x - data.parent_x) * data.scroll_x).round()
+                        / handle.cx.style().dpi_factor as f32;
+                    let top = ((data.child_y - data.parent_y) * data.scroll_y).round()
+                        / handle.cx.style().dpi_factor as f32;
+                    handle.left(Units::Pixels(-left.abs())).top(Units::Pixels(-top.abs()));
+                }
```
Comment (rhelmot) [inline core/src/views/scrollview.rs:118]: What does this fix?

File: core/src/views/textbox.rs [hunk 2]
```diff
@@ -336,18 +336,23 @@ impl Model for TextboxData {
...
             }
 
             TextEvent::EndEdit => {
                 self.edit = false;
+                cx.set_checked(false);
+                cx.release();
             }
 
-            TextEvent::Submit => {
+            TextEvent::Submit(flag) => {
```
Comment (rhelmot) [inline core/src/views/textbox.rs:353]: Can we pick a more descriptive name than "flag"?

File: core/src/views/scrollview.rs (hunk 1 above)
Comment (geom3trik) [reply core/src/views/scrollview.rs:118]: The cache contains values in physical coordinates but the style properties expect logical. This fixes an bug where extra space gets put at the bottom of a scrollable area.

**Review by rhelmot — APPROVED** [review]

**Pull request merged**